### ISSUE TITLE
Scalar widgets don't load in Safari (you get a no banana screen, though the no banana image also doesn't load)

### ISSUE THREAD

**Issue by lampholder** [vector-im/element-web#15405]
Widgets added with /addwidget work fine (modulo https://github.com/vector-im/element-web/issues/15404 which seems to stop them working on develop right now).

**Comment by jaywink** [comment]
I tried looking at this, but unfortunately the latest Safari that can be installed on Ubuntu is the Windows version from 2012, which isn't even able to load Element.

I think we need a developer with a Mac to look at this, anyone in Element Web team available? @dbkr ?

**Comment by jaywink** [comment]
(got access to a Mac)

@lampholder and/or @jryans could you confirm that turning off "Prevent cross-site tracking" makes the widgets show up? Or at least whether for you it was already turned off when seeing this error.

Also would be helpful to know if this was against scalar or scalar-staging.

![Selection_770](https://user-images.githubusercontent.com/1174866/95756789-ec1b7f80-0cae-11eb-9c5d-9a7d8a675a57.png)

Once turning that option off, I can load widgets on Safari 13.1.

**Comment by jryans** [comment]
@jaywink Yes, widgets load for me if I disable "Prevent cross-site tracking". It was previously enabled (by default) for me. I am testing with scalar-staging via develop.element.io.

**Comment by jryans** [comment]
(Or else redesign how widgets access cookies etc.)

**Comment by jryans** [comment]
We _might_ be able to get by here with limited changes if Scalar were to set it's access token via HTTP response instead of using `document.cookie`. It's hard to say anything with confidence though, as I have yet to find a clear summary of all the changes Safari has made.

**Comment by ara4n** [comment]
If we aren't going to fix this, please can we fail gracefully (i.e. route people to chrome / FF / electron) fairly urgently?

**Comment by jaywink** [comment]
@jryans I assume the "route to chrome / FF / electron" as above is something that would be done on Element side, right?

**Comment by t3chguy** [comment]
I don't think Element would be able to see that error happening inside the iframe due to cross-origin issues

**Comment by jryans** [comment]
Well, I suppose it depends what @ara4n has in mind... We could show a generic warning for all Scalar widgets in Safari saying they are not supported, I suppose?
